pennylane/templates/subroutines/hilbert_schmidt.py

```diff
@@ -105,6 +105,16 @@ def _flatten(self):
         )
         return self.data, metadata
 
+    @classmethod
+    def _primitive_bind_call(cls, *params, v_function, v_wires, u_tape, id=None):
+        # pylint: disable=arguments-differ
+        if qml.math.is_abstract(v_wires):
+            raise NotImplementedError(
+                "HilbertSchmidt does not support dynamic wires while using qml.capture."
+            )
```

Comment:
Is it some specific code in the template that is not traceable with wires?

Reply:
Basically hilbert-schmidt is special operator that we've decided not to deal with right now.  For example, it accepts a tape. The operations in the tape might be abstarct, but the tape itself won't be.

It also accepts a function, but it doesn't really make sense to have nested jaxpr here either.

My opinion is that if we want `HilbertSchmidt` to work with catalyst and capture, we should rewrite the call signature.

Reply:
This sounds similar to the control flow operators we have, like for loop. Could the tape be turned into jaxpr and attached as metadata to the operator (same as jax and catalyst do for primitives with a "body")?

I'm happy to delay this to a future PR though.

Reply:
created [a story](https://app.shortcut.com/xanaduai/story/64182/plxpr-can-capture-hilbertschmidt)